packages/scripts/src/Case.ts

```diff
@@ -0,0 +1,76 @@
+import z from "zod";
+
+// Map of embedding model name -> vector (array of numbers)
+export const Embeddings = z.record(z.string(), z.number().array());
+
+export type Embeddings = z.infer<typeof Embeddings>;
+
+export const PromptAndEmbeddings = z.object({
+  prompt: z.string(),
+  embeddings: Embeddings,
+});
+
+export type PromptAndEmbeddings = z.infer<typeof PromptAndEmbeddings>;
+
+export const RelevanceMetrics = z.object({
+  // normalized square magnitude difference (lower = closer = better)
+  norm_sq_mag_diff: z.number(),
+  // cosine similarity (are vectors pointing the same way?) [-1, 1]
+  cos_similarity: z.number(),
+});
+
+export type RelevanceMetrics = z.infer<typeof RelevanceMetrics>;
+
+export const ScoredPromptAndEmbeddings = PromptAndEmbeddings.and(
+  z.object({
+    relevance:
+      // embedding model name -> score
+      z.record(z.string(), RelevanceMetrics),
+  })
+);
+
+export type ScoredPromptAndEmbeddings = z.infer<
+  typeof ScoredPromptAndEmbeddings
+>;
+
+export const LlmAsJudgment = z
+  .object({
+    reasonableness: z.number(),
+    clarity: z.number(),
+    specificity: z.number(),
+    fit: z.number(),
+    assumption: z.number(),
+    impact: z.number(),
+    guidance: z.string(),
+  })
+  .partial();
```

Comment:
i recommend colocating this with the prompt, since if you're using structured outputs (as recommended below), this will become part of the prompt. you can also include more strict typing like .min() and .max(), plus .describe() for the model to see